### PR TITLE
Added `Router.resource` for defining a singular resource

### PR DESCRIPTION
**Pull request by joshnuss**
Adds the ability to define a set of routes that wont require an index action or an ID parameter in the path.

Example:

```elixir
# in web.router.ex
resource "/session", SessionsController
```

Will generate the following routes:

    * `GET /session` => `:show`
    * `GET /session/new` => `:new`
    * `POST /session` => `:create`
    * `GET /session/edit` => `:edit`
    * `PATCH /session` => `:update`
    * `PUT /session` => `:update`
    * `DELETE /session` => `:delete`

### CODE REVIEW

File: test/phoenix/router/resource_test.exs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    use Phoenix.Router
+
+    resource "/account", AccountController, alias: Api do
+      resources "/comments", CommentController do
+        get "/special", CommentController, :special
+      end
+      resource "/session", SessionController, except: [:delete]
+    end
+
+    resource "/session", Api.SessionController, only: [:show]
```
Comment (josevalim) [inline test/phoenix/router/resource_test.exs]: We can skip those routes and the ones below and their respective tests IMO. Just keeping it simple. :)

**Comment by josevalim** [comment]
:+1: This looks great. I think we can simplify the tests a bit and we need tests for the named helpers!

**Comment by joshnuss** [comment]
Groovy, will make those changes.

Do think we should have a seperate file for resource_test.exs and resources_test.exs? or should they be merged?

**Comment by josevalim** [comment]
Keeping them separated is fine imo.

**Comment by joshnuss** [comment]
Simplified the tests for resource (removed duplicate tests for `resources` stuff) and added tests for helpers too.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**